nifi-nar-bundles/nifi-asn1-bundle/nifi-asn1-services/src/main/java/org/apache/nifi/jasn1/JASN1Utils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.jasn1;
+
+import com.beanit.jasn1.ber.types.BerType;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import org.apache.nifi.util.Tuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.util.Arrays;
+
+class JASN1Utils {
+
+    private static Logger logger = LoggerFactory.getLogger(JASN1Utils.class);
+
+    /**
+     * Since the same class can be read many times in case reading an array, having a cache can reduce processing time.
+     * Class.getDeclaredMethod is time consuming.
+     */
+    private static final LoadingCache<Tuple<Class<? extends BerType>, String>, Method> getterCache = Caffeine.newBuilder()
+        .maximumSize(1000)
+        .build(JASN1Utils::getGetter);
+
+    private static Method getGetter(Tuple<Class<? extends BerType>, String> methodKey) {
+        try {
+            return methodKey.getKey().getDeclaredMethod(methodKey.getValue());
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Failed to get method from " + methodKey, e);
```

Comment:
While doing some test I found that this could benefit from a more descriptive error message (as the `methodKey` doesn't have good `toString`.)
This worked for me very well:
`throw new RuntimeException("Method '" + methodName + "' not found in '" + clazz.getSimpleName() + "'", e);`